linker/shared/src/test/scala/org/scalajs/linker/BackwardsCompatTest.scala

```diff
@@ -109,6 +109,34 @@ class BackwardsCompatTest {
     test(classDefs, MainTestModuleInitializers)
   }
 
+  @Test
+  def testArrayNewInstanceHacks_Issue5107(): AsyncResult = await {
```

Comment:
The reference to 5107 is probably reasonable, but then maybe we should change its title and the title of this PR?